In [6]:
from sklearn.datasets import load_iris
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier


In [7]:
iris = load_iris()

In [8]:
dir(iris)

['DESCR',
 'data',
 'data_module',
 'feature_names',
 'filename',
 'frame',
 'target',
 'target_names']

In [10]:
from sklearn.model_selection import cross_val_score

**1. Logistic Regression**

In [49]:
l_score = cross_val_score(LogisticRegression(random_state=42, max_iter=200), iris.data, iris.target)
l_score

array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

In [50]:
from numpy import average
average(l_score)

np.float64(0.9733333333333334)

**2. SVM**

In [40]:
svm_score = cross_val_score(SVC(gamma=1, kernel='linear', C=1), iris.data, iris.target)
svm_score

array([0.96666667, 1.        , 0.96666667, 0.96666667, 1.        ])

In [41]:
average(svm_score)

np.float64(0.9800000000000001)

In [52]:
svm_score = cross_val_score(SVC(gamma=1, kernel='poly', C=1), iris.data, iris.target)
svm_score

array([0.96666667, 0.96666667, 0.93333333, 0.93333333, 1.        ])

In [53]:
average(svm_score)

np.float64(0.96)

**3. Random Forest**

In [62]:
rf_score = cross_val_score(RandomForestClassifier(n_estimators=40,), iris.data, iris.target)
rf_score

array([0.96666667, 0.96666667, 0.93333333, 0.93333333, 1.        ])

In [63]:
average(rf_score)

np.float64(0.96)

**Decision tree**

In [64]:
d_score = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
d_score

array([0.96666667, 0.96666667, 0.9       , 0.96666667, 1.        ])

In [65]:
average(d_score)

np.float64(0.9600000000000002)

**Best score so far is from SVM: 0.9800000000000001**


*   SVM > Logistic Regression > Decision Tree ~ Random Forest


